### PR TITLE
Created InfixtoPrefix program in Java

### PR DESCRIPTION
**Pull request by iamvs-2002**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description
Created program to convert an expression from Infix to Prefix in Java
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#2970 
(Write your answer here.)

### CODE REVIEW

**Comment by iamvs-2002** [comment]
@ankitaggarwal23 Hy. Why does Java(beta) check fail even when the code is completely fine? Please help.

**Comment by atarax665** [comment]
> @ankitaggarwal23 Hy. Why does Java(beta) check fail even when the code is completely fine? Please help.

You can ignore it, there is some glitch at deepsource's end.

File: Java/other/InfixtoPrefix.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
+/**
+ * Infix to Prefix Expression
+ * 
+ * Infix Expression : A * B + C / D
```
Comment (atarax665) [inline Java/other/InfixtoPrefix.java]: Remove sample I/O from here as you have already added them at the end.

File: Java/other/InfixtoPrefix.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+ * Prefix Expression : + * A B/ C D
+ * 
+ * Infix Expression : A+B*C
+ * Prefix Expression : +A*BC
+ * 
+ * Algorithm Used:
+ * First the gicen infix expression is reversed.
```
Comment (atarax665) [inline Java/other/InfixtoPrefix.java]:
```suggestion
 * First the given infix expression is reversed.
```

File: Java/other/InfixtoPrefix.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        for (i = result.length()-1; i >=0 ; i--) { 
+            arr[result.length()-i-1] = result.charAt(i);
+        }
+        //finally, arr contains the prefix expression.
+        return String.valueOf(arr);
+    }
+}
+/**
+ * Sample Outputs:
+ *
```
Comment (atarax665) [inline Java/other/InfixtoPrefix.java:136]: Add space and time complexity of your approach.

File: Java/other/InfixtoPrefix.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            }
+        }
+        //the result string is now a postfix expression
+        //reversing the string to obtain the prefix expression.
+        //converting the final result string to a char array
+        char[] arr = new char[result.length()];
+        for (i = result.length()-1; i >=0 ; i--) { 
+            arr[result.length()-i-1] = result.charAt(i);
+        }
+        //finally, arr contains the prefix expression.
```
Comment (atarax665) [inline Java/other/InfixtoPrefix.java]: Remove trivial comments.

File: Java/other/InfixtoPrefix.java [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+ * Kindly enter the infix expression: 
+ * x+y*z/w+u
+ * Prefix Expression: ++x/*yzwu
+ * 
+ * Kindly enter the infix expression: 
+ * A * B + C / D
+ * Prefix Expression: +*A  B / C  D
+ * 
+ */
+
```
Comment (atarax665) [inline Java/other/InfixtoPrefix.java]: Remove this line.

**Review by atarax665 — CHANGES_REQUESTED** [review]
The code looks good to merge, just make all the suggested changes.

**Comment by iamvs-2002** [comment]
> > @ankitaggarwal23 Hy. Why does Java(beta) check fail even when the code is completely fine? Please help.
> 
> You can ignore it, there is some glitch at deepsource's end.

Ohh okay, thanks.

File: Java/other/InfixtoPrefix.java [hunk 6]
```diff
@@ -0,0 +1,151 @@
+/**
+ * Infix to Prefix Expression
+ * 
+ * Algorithm Used:
+ * First the given infix expression is reversed.
+ * By reversing, every '(' becomes ')' and vice versa.
+ * Then, we obtain the postfix of the reversed expression.
+ * Now, we reverse this postfix expression to obtain the Prefix expression.
+ * 
+ * Time Complexity: O(n^2)
```
Comment (atarax665) [inline Java/other/InfixtoPrefix.java]: Shift time and space complexity below the sample input/output.

**Review by atarax665 — APPROVED** [review]

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**